Server/mods/deathmatch/logic/CMapEventManager.cpp

```diff
@@ -341,18 +342,86 @@ bool CMapEventManager::HandleExists(CLuaMain* pLuaMain, const char* szName, cons
     return false;
 }
 
-void CMapEventManager::AddInternal(CMapEvent* pEvent)
+void CMapEventManager::DeleteInternal(CMapEvent* pMapEvent)
 {
+    SEvent* pEvent = g_pGame->GetEvents()->Get(pMapEvent->GetName());
+
+    // For now we only care about CURSOR_EVENT
+    if (pEvent && pEvent->eServerRPCFunction == eServerRPCFunctions::CURSOR_EVENT)
+    {
+        bool bFound = !GetHandlesByServerRPCFunction(pEvent->eServerRPCFunction).empty();
+        if (!bFound)
+        {
+            bFound = !g_pGame->GetMapManager()->GetRootElement()->GetEventManager()->GetHandlesByServerRPCFunction(pEvent->eServerRPCFunction).empty();
+
+            if (!bFound)
+            {
+                auto iter = g_pGame->GetMapManager()->GetRootElement()->IterBegin();
+                for (; iter != g_pGame->GetMapManager()->GetRootElement()->IterEnd(); iter++)
+                {
+                    bFound = !(*iter)->GetEventManager()->GetHandlesByServerRPCFunction(pEvent->eServerRPCFunction).empty();
+                    if (bFound)
+                        break;
+                }
+
+                // Let players know
+                if (!bFound)
+                {
+                    std::map<eServerRPCFunctions, bool> map;
+                    map.insert(std::make_pair(pEvent->eServerRPCFunction, true));
+                    CServerRPCControlPacket Packet(map);
+                    g_pGame->GetPlayerManager()->BroadcastOnlyJoined(Packet);
+                }
+            }
+        }
+    }
+
+    delete pMapEvent;
+}
+
+void CMapEventManager::AddInternal(CMapEvent* pMapEvent)
+{
+    SEvent* pEvent = g_pGame->GetEvents()->Get(pMapEvent->GetName());
+
+    // For now we only care about CURSOR_EVENT
+    if (pEvent && pEvent->eServerRPCFunction == eServerRPCFunctions::CURSOR_EVENT)
+    {
+        bool bFound = !GetHandlesByServerRPCFunction(pEvent->eServerRPCFunction).empty();
+        if (!bFound)
+        {
+            bFound = !g_pGame->GetMapManager()->GetRootElement()->GetEventManager()->GetHandlesByServerRPCFunction(pEvent->eServerRPCFunction).empty();
+
+            if (!bFound)
+            {
+                auto iter = g_pGame->GetMapManager()->GetRootElement()->IterBegin();
+                for (; iter != g_pGame->GetMapManager()->GetRootElement()->IterEnd(); iter++)
+                {
+                    bFound = !(*iter)->GetEventManager()->GetHandlesByServerRPCFunction(pEvent->eServerRPCFunction).empty();
+                    if (bFound)
+                        break;
+                }
+
+                // Let players know
+                if (!bFound)
+                {
+                    std::map<eServerRPCFunctions, bool> map;
+                    map.insert(std::make_pair(pEvent->eServerRPCFunction, false));
+                    CServerRPCControlPacket Packet(map);
+                    g_pGame->GetPlayerManager()->BroadcastOnlyJoined(Packet);
+                }
+            }
+        }
+    }
+
```

Comment:
These look identical besides the bool in map.insert. Should this be put into a separate "toggle" function instead that takes a boolean? And you can return in there instead of if's